twake/backend/node/src/services/preview/src/previews/services/image.ts

```diff
@@ -0,0 +1,17 @@
+import sharp from "sharp";
+import { unlink } from "fs/promises";
+
+export async function generatePreview(inputPath: string, outputPath: string, outputFormat: string) {
+  outputFormat = "png";
+
+  await sharp(inputPath)
+    .resize(300, 200)
+    .toFile(`${outputPath}.${outputFormat}`, function (err) {});
```

Comment:
This is strange to have `await` and this error callback. Your sync issue probably comes from here